### ISSUE TITLE
Plugin does not correctly handle bind mounts when BuildKit is enabled or when the source is the context not another Image.

### ISSUE THREAD

**Issue by nigelgbanks** [Islandora-Devops/isle-gradle-docker-plugin#4]
Currently the plugin adds `COPY` commands when for bind mounts always, it should only do this in the case when **BuildKit is disabled**. 

Additionally it should distinguish between contextual bind mounts and bind mounts from another image as the syntax of the `COPY` command changes depending on the source type (contextual, bind). At the moment contextual bind mounts are not supported by this plugin.

See the documentation for [bind mounts](https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/syntax.md#run---mounttypebind-the-default-mount-type)

Currently using the following syntax (where extensions is a folder in the docker build context):

```Dockerfile
RUN --mount=type=bind,from=extensions,source=.,target=/extensions
```

Will get converted into

```Dockerfile
COPY --from=extensions . /extensions
```

Which will fail as the build system will be looking  for the Docker image `extensions` producing an error like the following:

```bash
ERROR: pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
```

The generated copy should look like so:

```Dockerfile
COPY extensions/  /extensions/
```

The generation of the `COPY` instruction will become a bit more complex in that it will need to account for relative paths the current version does not as when specifying `source` for images absolute paths are required as there is no relative path. This is not the case with bind mounts of the Docker build context.

**Comment by nigelgbanks** [comment]
Note that changing this we can still have an unavoidable bug in that if a `FROM` alias has the same name as a folder in the Drupal context we can't distinguish between the two and what is intended.

**Comment by nigelgbanks** [comment]
> Note that changing this we can still have an unavoidable bug in that if a `FROM` alias has the same name as a folder in the Drupal context we can't distinguish between the two and what is intended.

If I'm reading the docs correctly in that case the user should drop the `from=value` to indicate that it from should be the build context to distinguish between the two so there is a way to make this clear.

**Comment by nigelgbanks** [comment]
Resolved with version 0.0.3
